link/src/main/java/com/stripe/android/link/ui/PrimaryButton.kt

```diff
@@ -137,25 +138,44 @@ internal fun PrimaryButton(
                     )
                 }
             }
-            // Show icon only when button label is visible
-            if (icon != null &&
-                state in setOf(PrimaryButtonState.Enabled, PrimaryButtonState.Disabled)
-            ) {
-                Icon(
-                    painter = painterResource(id = icon),
-                    contentDescription = null,
-                    modifier = Modifier
-                        .height(16.dp)
-                        // width should be 13dp and must include the horizontal padding
-                        .width(13.dp + 40.dp)
-                        .padding(horizontal = HorizontalPadding),
-                    tint = MaterialTheme.linkColors.buttonLabel.copy(alpha = LocalContentAlpha.current)
-                )
+            // Show icons only when button label is visible
+            if (state in setOf(PrimaryButtonState.Enabled, PrimaryButtonState.Disabled)) {
+                iconStart?.let { icon ->
+                    Box(
+                        modifier = Modifier.fillMaxSize(),
+                        contentAlignment = Alignment.CenterStart
+                    ) {
+                        PrimaryButtonIcon(icon)
+                    }
+                }
+                iconEnd?.let { icon ->
+                    Box(
+                        modifier = Modifier.fillMaxSize(),
+                        contentAlignment = Alignment.CenterEnd
+                    ) {
+                        PrimaryButtonIcon(icon)
+                    }
+                }
             }
         }
     }
 }
 
+@Composable
+private fun PrimaryButtonIcon(
+    @DrawableRes icon: Int
+) {
+    Icon(
+        painter = painterResource(id = icon),
+        contentDescription = null,
+        modifier = Modifier
+            .height(16.dp)
+            .width(13.dp + 40.dp)
```

Comment:
Similar comment here: Can we use the intrinsic size and just vertically center the content?

Reply:
The icons might have different sizes so we want to scale them to a fixed size. Removed padding now that they're inside the button.